src/test/unit/test_tracer.py

```diff
@@ -228,6 +231,51 @@ def lambda_test_function(event, context):
         assert http_spans[0]["info"]["httpInfo"]["request"]["headers"]
 
 
+@pytest.mark.parametrize("propagate_w3c", [True, False])
+def test_add_w3c_headers_to_http_without_headers(monkeypatch, context, propagate_w3c, aws_env):
+    @lumigo_tracer(propagate_w3c=propagate_w3c)
+    def lambda_test_function(event, context):
+        conn = http.client.HTTPConnection("www.google.com")
+        conn.request("POST", "/", json.dumps({"a": "b"}))
+        return {"hello": "world"}
+
+    lambda_test_function({}, context)
+    http_spans = list(SpansContainer.get_span().spans.values())
+    assert (
+        TRACEPARENT_HEADER_NAME in http_spans[0]["info"]["httpInfo"]["request"]["headers"]
+    ) == propagate_w3c
```

Comment:
To this tests and all the other related tests, can you also verify:
1. The value of `http_spans[0]["info"]["httpInfo"]["request"]["headers"]`.
2. That `conn.request` was called with all the correct argument.

Reply:
Take a look! This is an interesting test 🤔